include/nbl/video/CCUDADevice.h

```diff
@@ -179,16 +179,51 @@ class CCUDADevice : public core::IReferenceCounted
 		static CUresult acquireAndGetMipmappedArray(GraphicsAPIObjLink<video::IGPUImage>* linksBegin, GraphicsAPIObjLink<video::IGPUImage>* linksEnd, CUstream stream);
 		static CUresult acquireAndGetArray(GraphicsAPIObjLink<video::IGPUImage>* linksBegin, GraphicsAPIObjLink<video::IGPUImage>* linksEnd, uint32_t* arrayIndices, uint32_t* mipLevels, CUstream stream);
 #endif
+		CUdevice getInternalObject() const { return m_handle; }
+		const CCUDAHandler* getHandler() const { return m_handler.get();  }
 
+		struct SSharedCUDAMemory
+		{
+			size_t size;
+			CUdeviceptr ptr;
+			CUmemGenericAllocationHandle memory;
+			void* osHandle;
+		};
+
+		core::smart_refctd_ptr<IGPUBuffer> exportGPUBuffer(SSharedCUDAMemory mem, ILogicalDevice* device);
+
+		CUresult importGPUBuffer(IGPUBuffer* buf, SSharedCUDAMemory* outPtr);
+		CUresult createExportableMemory(size_t size, size_t alignment, SSharedCUDAMemory* outMem);
+		CUresult releaseExportableMemory(SSharedCUDAMemory mem);
 	protected:
+
+		struct SCUDACleaner : video::ICleanup, SSharedCUDAMemory
```

Comment:
please use composition over inheritance, there's no reason to inherit from `SSharedCUDAMemoery`